modules/apps/portal/portal-cluster-multiple/src/main/java/com/liferay/portal/cluster/multiple/internal/jgroups/JGroupsClusterChannelFactory.java

```diff
@@ -220,6 +238,81 @@ protected void setProps(Props props) {
 		_props = props;
 	}
 
+	private InputStream _getConfigStream(String configLocation)
+		throws IOException {
+
+		InputStream configStream = ConfiguratorFactory.getConfigStream(
+			configLocation);
+
+		if (configStream != null) {
+			return configStream;
+		}
+
+		ClassLoader classLoader = PortalClassLoaderUtil.getClassLoader();
+
+		return classLoader.getResourceAsStream(configLocation);
+	}
+
+	private ProtocolStackConfigurator _parseConfigFile(String configLocation)
+		throws Exception {
+
+		try (InputStream inputStream = _getConfigStream(configLocation)) {
+			if (inputStream == null) {
+				throw new FileNotFoundException(
+					"Config file not found: ".concat(configLocation));
+			}
+
+			String configXML = StreamUtil.toString(inputStream);
+
+			int index = 0;
+
+			StringBundler sb = new StringBundler();
+
+			while (index < configXML.length()) {
+				int startIndex = configXML.indexOf(
+					StringPool.CLOSE_CURLY_BRACE, index);
```

Comment:
@dantewang  This line is wrong, we should check DOLLAR_AND_OPEN_CURLY_BRACE instead of CLOSE_CURLY_BRACE.

Could you please fix and resned?

Thanks.
Tina